src/sentry/models/team.py

```diff
@@ -280,14 +285,22 @@ def transfer_to(self, organization):
         ).delete()
 
         if new_team != self:
+            # Delete the old team
             cursor = connections[router.db_for_write(Team)].cursor()
             # we use a cursor here to avoid automatic cascading of relations
             # in Django
             try:
                 cursor.execute("DELETE FROM sentry_team WHERE id = %s", [self.id])
+                cursor.execute("DELETE FROM sentry_actor WHERE team_id = %s", [new_team.id])
             finally:
                 cursor.close()
 
+            # Change whatever new_team's actor is to the one from the old team.
+            with transaction.atomic():
+                Actor.objects.filter(id=self.actor_id).update(team_id=new_team.id)
+                new_team.actor_id = self.actor_id
+                new_team.save()
```

Comment:
Should we have a test for this as re-using the actor is something we could overlook in the future.